src/test/java/net/snowflake/client/jdbc/PreparedStatement1IT.java

```diff
@@ -790,4 +791,46 @@ public void manualTestForPreparedStatementLogging() throws SQLException {
       }
     }
   }
+
+  /**
+   * This test shows how to use batch requests with variant columns. Syntax <code>
+   * insert into ... Select column1, ... from values (?,...)</code> is crucial there.
+   * CLIENT_STAGE_ARRAY_BINDING_THRESHOLD parameter is used on session to set the minimal number of
+   * bind parameters to create stage file with provided parameters.
+   */
+  @Test
+  public void shouldUseVariantBindParametersSNOW1531307() throws SQLException {
+    try (Connection con = getConnection();
+        Statement stmt = con.createStatement()) {
+      String tableName = "SNOW1531307";
+      stmt.execute(
+          "CREATE OR REPLACE TABLE " + tableName + "(id text, json1 variant, json2 variant)");
+      executePreparedStatementWithBatches(con, tableName, 20);
+      try {
+        stmt.execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 10");
```

Comment:
Does it always work? I think it works only on envs where we have admin permissions.

Reply:
tested and unfortunately you are right - we will run this test only on jenkins